go-controller/pkg/ovn/kubevirt.go

```diff
@@ -56,18 +64,300 @@ func (oc *DefaultNetworkController) deleteDHCPOptions(pod *kapi.Pod) error {
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureARPProxyContainsSubnet(switchName string, subnet *net.IPNet) error {
+	lspName := ovntypes.SwitchToRouterPrefix + switchName
+	lsp, err := libovsdbops.GetLogicalSwitchPort(oc.nbClient, &nbdb.LogicalSwitchPort{Name: lspName})
+	if err != nil {
+		return err
+	}
+	arpProxy, ok := lsp.Options["arp_proxy"]
+	if !ok {
+		return fmt.Errorf("missing arp_proxy at logical switch port '%s'", lspName)
+	}
+	subnetString := subnet.String()
+	for _, arpProxyEntry := range strings.Split(arpProxy, " ") {
+		if arpProxyEntry == subnetString {
+			return nil
+		}
+	}
+	arpProxy += " " + subnetString
+	lsp.Options["arp_proxy"] = arpProxy
+	if err := libovsdbops.UpdateLogicalSwitchPortSetOptions(oc.nbClient, lsp); err != nil {
+		return fmt.Errorf("failed update logical switch port to add subnet to arp_proxy: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) removeARPProxySubnet(switchName string, subnet *net.IPNet) error {
+	lspName := ovntypes.SwitchToRouterPrefix + switchName
+	lsp, err := libovsdbops.GetLogicalSwitchPort(oc.nbClient, &nbdb.LogicalSwitchPort{Name: lspName})
+	if err != nil {
+		return err
+	}
+	arpProxy, ok := lsp.Options["arp_proxy"]
+	if !ok {
+		return fmt.Errorf("missing arp_proxy at logical switch port '%s'", lspName)
+	}
+	subnetString := subnet.String()
+	arpProxyProcessed := []string{}
+	for _, arpProxyEntry := range strings.Split(arpProxy, " ") {
+		if arpProxyEntry == subnetString {
+			continue
+		}
+		arpProxyProcessed = append(arpProxyProcessed, arpProxyEntry)
+	}
+	lsp.Options["arp_proxy"] = strings.Join(arpProxyProcessed, " ")
+	if err := libovsdbops.UpdateLogicalSwitchPortSetOptions(oc.nbClient, lsp); err != nil {
+		return fmt.Errorf("failed update logical switch port to remove subnet from arp_proxy: %v", err)
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) ensureRoutingForVM(pod *kapi.Pod) error {
+	isLiveMigrationLefover, err := kubevirt.PodIsLiveMigrationLeftOver(oc.client, pod)
+	if err != nil {
+		return err
+	}
+	if util.PodWantsHostNetwork(pod) || !kubevirt.AllowPodBridgeNetworkLiveMigration(pod.Annotations) || isLiveMigrationLefover {
+		return nil
+	}
+
+	targetNode := pod.Labels[kubevirt.NodeNameLabel]
+	targetStartTimestamp := pod.Annotations[kubevirt.MigrationTargetStartTimestampAnnotation]
+	// No live migration or target node was reached || qemu is already ready
+	if targetNode == pod.Spec.NodeName || targetStartTimestamp != "" {
+		if err := oc.enroutePodAddressesToNode(pod); err != nil {
+			return fmt.Errorf("failed enroutePodAddressesToNode for %s/%s: %w", pod.Namespace, pod.Name, err)
+		}
+		if err := oc.addVMSubnetsToARPProxy(pod); err != nil {
+			return fmt.Errorf("failed addVMSubnetsToARPProxy for %s/%s: %w", pod.Namespace, pod.Name, err)
+		}
+	}
+	return nil
+}
+
+func (oc *DefaultNetworkController) lrpAddresses(lrpName string) (string, string, error) {
```

Comment:
There must already exist a mechanism to do this but I am sure what is it. Tim will know.

Reply:
Have found:
- util.GetLRPAddrs
- util.MatchFirstIPNetFamily

Will use them.